framework/include/materials/MaterialData.h

```diff
@@ -169,6 +181,10 @@ class MaterialData
   /// Status of storage swapping (calling swap sets this to true; swapBack sets it to false)
   bool _swapped;
 
+  /// Use non-destructive resize of material data (calling resize() will not reduce size).
+  /// Default is false (normal resize behaviour)
+  mutable bool _resize_only_if_smaller;
```

Comment:
Why a mutable here? I don't believe MaterialData is constant in the context you are using it nor is the place where you are calling your new onlyResizeIfSmaller() method. Am I missing something?

Reply:
I thought that `_material_data` was a const reference but it clearly isn't, so it probably doesn't need to be mutable. I'll remove it